### PR TITLE
refactor: Use logdown for logging

### PR DESCRIPTION
**Pull request by pwnsdx**

### CODE REVIEW

File: electron/package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "image-type": "3.0.0",
     "jsrsasign": "8.0.12",
     "lodash": "4.17.11",
+    "logdown": "3.2.7",
```
Comment (ffflorian) [inline electron/package.json]: Let's also remove `debug` from the dependencies.

File: electron/src/main.ts [hunk 2]
```diff
@@ -328,21 +331,22 @@ const renameLogFile = () => {
...
 class ElectronWrapperInit {
-  debug: debug.IDebugger;
+  logger: logdown.Logger;
 
   constructor() {
-    this.debug = debug('ElectronWrapperInit');
+    this.logger = logdown('wire-desktop/main.ts', {
+      logger: console,
+      markdown: false,
+    });
```
Comment (ffflorian) [inline electron/src/main.ts]: Why not use the logger from line 32?

File: electron/src/main.ts [hunk 3]
```diff
@@ -387,7 +391,7 @@ class ElectronWrapperInit {
             // Verify the URL being loaded
             if (!util.isMatchingHost(_url, BASE_URL)) {
               event.preventDefault();
-              webviewProtectionDebug('Prevented to show an unauthorized <webview>. URL: %s', _url);
+              this.logger.log('Prevented to show an unauthorized <webview>. URL: %s', _url);
```
Comment (ffflorian) [inline electron/src/main.ts]:
```suggestion
              this.logger.log(`Prevented to show an unauthorized <webview>. URL: ${_url}`);
```

File: electron/src/main.ts [hunk 4]
```diff
@@ -356,16 +360,16 @@ class ElectronWrapperInit {
...
     };
 
     const willNavigateInWebview = (event: Event, _url: string) => {
       // Ensure navigation is to a whitelisted domain
       if (util.isMatchingHost(_url, BASE_URL)) {
-        webviewProtectionDebug('Navigating inside webview. URL: %s', _url);
+        this.logger.log('Navigating inside webview. URL: %s', _url);
       } else {
-        webviewProtectionDebug('Preventing navigation inside webview. URL: %s', _url);
+        this.logger.log('Preventing navigation inside webview. URL: %s', _url);
```
Comment (ffflorian) [inline electron/src/main.ts]:
```suggestion
        this.logger.log(`Preventing navigation inside webview. URL: ${_url}`);
```

File: electron/src/main.ts [hunk 5]
```diff
@@ -356,16 +360,16 @@ class ElectronWrapperInit {
...
-      webviewProtectionDebug('Opening an external window from a webview. URL: %s', url);
+      this.logger.log('Opening an external window from a webview. URL: %s', url);
       return shell.openExternal(url);
     };
 
     const willNavigateInWebview = (event: Event, _url: string) => {
       // Ensure navigation is to a whitelisted domain
       if (util.isMatchingHost(_url, BASE_URL)) {
-        webviewProtectionDebug('Navigating inside webview. URL: %s', _url);
+        this.logger.log('Navigating inside webview. URL: %s', _url);
```
Comment (ffflorian) [inline electron/src/main.ts]:
```suggestion
        this.logger.log(`Navigating inside webview. URL: ${_url}`);
```

File: electron/src/main.ts [hunk 6]
```diff
@@ -356,16 +360,16 @@ class ElectronWrapperInit {
         return new SingleSignOn(main, event, url, options).init();
       }
 
-      webviewProtectionDebug('Opening an external window from a webview. URL: %s', url);
+      this.logger.log('Opening an external window from a webview. URL: %s', url);
```
Comment (ffflorian) [inline electron/src/main.ts]:
```suggestion
      this.logger.log(`Opening an external window from a webview. URL: ${url}`);
```

File: electron/src/main.ts (hunk 2 above)
Comment (bennyn) [reply electron/src/main.ts]: because it's a different class logging

Comment (ffflorian) [reply electron/src/main.ts]: But it has the same name in the logger..?

Comment (bennyn) [reply electron/src/main.ts]: yes, that should be changed

**Pull request merged**